packages/affiliates/test/mocha/dappmining.js

```diff
@@ -0,0 +1,230 @@
+const Path = require("path");
+const assert = require("assert");
+const highland = require("highland");
+
+const { DappMining } = require("../../libs/affiliates");
+const { getWeb3 } = require("@uma/common");
+const { getAbi } = require("@uma/core");
+const { mocks } = require("../../libs/datasets");
+const { SharedAttributions, Balances } = require("../../libs/models");
+const { EmpAttributions } = require("../../libs/processors");
+const { Queries } = mocks;
+const { encodeAttribution, EncodeCallData } = require("../../libs/contracts");
+
+const web3 = getWeb3();
+const { fromWei, toWei } = web3.utils;
+const empAbi = getAbi("ExpiringMultiParty");
+const datasetPath = Path.join(__dirname, "../datasets/dapp-mining-set1");
+const config = require("../datasets/dapp-mining-set1/config.json");
+
+const encodeCallData = EncodeCallData(empAbi);
+async function makeTx(blockNumber, collateral, tokens, user, tag) {
+  return {
+    blockNumber,
+    name: "create",
+    from_address: user,
+    args: [collateral, tokens],
+    input: encodeAttribution(encodeCallData("create", [collateral.toString()], [tokens.toString()]), tag)
+  };
+}
+
+describe("DappMining", function() {
```

Comment:
This starts the unit tests for the dapp mining logic. In here several functions are tested standalone and at the end one integration test using a downloaded (and unmodified) dataset with tagged and untagged transactions.